crd/apis/network/v1/gkenetworkparams_types.go

```diff
@@ -0,0 +1,73 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:storageversion
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GKENetworkParams represent GKE specific parameters for the network.
+type GKENetworkParams struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   GKENetworkParamsSpec   `json:"spec,omitempty"`
+	Status GKENetworkParamsStatus `json:"status,omitempty"`
+}
+
+// DeviceModeType defines mode in which the devices will be used by the Pod
+// +kubebuilder:validation:Enum=DPDKUIO;DPDKVFIO;NetDevice
```

Comment:
@naveennathan

I believe this has to be the string value in order to fix the issue weilong brought up.